Performance introspection

ludv1x opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

ludv1x committed:
Introduced ThreadStatus. [#CLICKHOUSE-2910]

ludv1x committed:
Threads and ProfileEvents in system.processes. [#CLICKHOUSE-2910]

ludv1x committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

ludv1x committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

ludv1x committed:
Add profile events and settings logging. [#CLICKHOUSE-2910]

ludv1x committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

ludv1x committed:
Add more perfomance counters. [#CLICKHOUSE-2910]

ludv1x committed:
Refactoring. [#CLICKHOUSE-2910]

ludv1x committed:
[WIP] Introduced query_thread_log. [#CLICKHOUSE-2910]

ludv1x committed:
Better AsynchronousBlockInputStream in sync mode. [#CLICKHOUSE-2910]

ludv1x committed:
Initialize external tables after enqueue query. [#CLICKHOUSE-2910]

ludv1x committed:
Simplified code via introducing states. [#CLICKHOUSE-2910]

ludv1x committed:
C++ify taskstats getter, periodic auto update of profile events. [#CLICKHOUSE-2910]

ludv1x committed:
Add server logs passing to client. [#CLICKHOUSE-2910]

ludv1x committed:
Add more options for loggig. [#CLICKHOUSE-2910]

ludv1x committed:
Fixed several bug. Trun on --server_logs_level=warning in tests. [#CLICKHOUSE-2910]

Fixed several test to be compatible with --server_logs_level=warning.

ludv1x committed:
Fixed memory tracking. [#CLICKHOUSE-2910]

ludv1x committed:
Fixed exception safety in TCPHandler. [#CLICKHOUSE-2910]

ludv1x committed:
Add test for profile events logging. [#CLICKHOUSE-2910]

ludv1x committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

ludv1x committed:
Add CAP_NET_ADMIN capability check. [#CLICKHOUSE-2910]

ludv1x committed:
Disable compression for Log packets. [#CLICKHOUSE-2910]

alexey-milovidov commented on dbms/src/Common/TaskStatsInfoGetter.cpp:
```diff
@@ -242,5 +236,33 @@ int TaskStatsInfoGetter::getCurrentTID()
+    /// 0 - wasn't checked
+    /// 1 - checked, has no permissions
+    /// 2 - checked, has permissions
+    static std::atomic<int> premissions_check_status{0};
```
You can do check inside constructor of function-static variable and it will be thread safe automatically.

alexey-milovidov commented on dbms/src/Common/TaskStatsInfoGetter.cpp:
```diff
@@ -242,5 +236,33 @@ int TaskStatsInfoGetter::getCurrentTID()
+        premissions_check_status.store(status, std::memory_order_relaxed);
+    }
+
+    return status == 2;
```
Possible race condition.

ludv1x committed:
Add query_id and host_name to server logs. [#CLICKHOUSE-2910]

ludv1x committed:
Print query_id in logs. Add own logging interface. [#CLICKHOUSE-2910]

ludv1x committed:
Minor enhancement. [#CLICKHOUSE-2910]

ludv1x committed:
Add logs forwarding from remote severs to client. [#CLICKHOUSE-2910]

ludv1x committed:
Rewritten ThreadStatus via adding shared thread state. [#CLICKHOUSE-2910]

Fixed race condition in SystemLog.

ludv1x committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

ludv1x committed:
Fixed failures in utils, more precise test. [#CLICKHOUSE-2910]

ludv1x committed:
clickhouse_common_io does not depend on dbms. [#CLICKHOUSE-2910]

Get rid of extra linking dependencies for clickhouse_common_io. [#CLICKHOUSE-2910]

ludv1x commented on dbms/src/Core/Defines.h:
```diff
@@ -61,6 +61,7 @@
 #define DBMS_MIN_REVISION_WITH_TABLES_STATUS 54226
 #define DBMS_MIN_REVISION_WITH_TIME_ZONE_PARAMETER_IN_DATETIME_DATA_TYPE 54337
 #define DBMS_MIN_REVISION_WITH_SERVER_DISPLAY_NAME 54372
+#define DBMS_MIN_REVISION_WITH_SERVER_LOGS 54386
```
Don't forget to increment it after before release.

alexey-milovidov committed:
Merge branch 'master' into CLICKHOUSE-2910

alexey-milovidov committed:
Update Defines.h

alexey-milovidov committed:
Fixed setting descriptions #2482

alexey-milovidov committed:
Auto version update to [54389]

alexey-milovidov committed:
Leave thread number in log in a consistent place #2482

alexey-milovidov commented:
`server_logs_level` is non obvious name.
The user may think that it will affect the log level directly on server.
Will rename...

alexey-milovidov committed:
Renamed setting "send_logs_level" #2482

ludv1x commented on dbms/programs/server/TCPHandler.cpp:
```diff
@@ -173,8 +191,11 @@ void TCPHandler::runImpl()
 
-            sendEndOfStream();
+            /// Reset BlockIO in advance to log destruction actions
+            state.io.reset();
```
There is a problem with this invocation.
If an exception is thrown blow this line, then we will become here https://github.com/yandex/ClickHouse/pull/2482/files#diff-8389b5ff47b68bdd1959bf673e944fb4R203 and call onException() which will use destroyed object state.io.

alexey-milovidov replied:
Thanks! First I will simply remove it.

alexey-milovidov committed:
Little better #2482

alexey-milovidov committed:
Merge branch 'master' into CLICKHOUSE-2910

alexey-milovidov committed:
Fixed typo #2482

alexey-milovidov committed:
Merge branch 'master' into CLICKHOUSE-2910

alexey-milovidov committed:
Removed static std::string variable #2482

alexey-milovidov committed:
Merging #2482

alexey-milovidov committed:
Merging #2482

alexey-milovidov committed:
Merging #2482

alexey-milovidov committed:
Merging #2482

alexey-milovidov committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

alexey-milovidov committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

alexey-milovidov committed:
Miscellaneous #2482

alexey-milovidov committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

Conflicts:
	dbms/src/Interpreters/Settings.h
	dbms/tests/queries/0_stateless/00687_top_and_offset.sql

alexey-milovidov committed:
Removed duplicate query_id logging #2482

alexey-milovidov committed:
Auto version update to [18.11.0] [54406]

alexey-milovidov committed:
Merge branch 'CLICKHOUSE-2910' of github.com:yandex/ClickHouse into CLICKHOUSE-2910

alexey-milovidov committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

Conflicts:
	dbms/src/Common/ODBCBridgeHelper.h

alexey-milovidov committed:
Miscellaneous #2482

alexey-milovidov committed:
Merge remote-tracking branch 'origin/master' into CLICKHOUSE-2910

alexey-milovidov committed:
Removed bad code #2864

alexey-milovidov committed:
Added clarifying comment [#CLICKHOUSE-2]
